service-template-node update and eslint fixes

ottomata opened the pull request:
Swagger has been upgraded to openapi 3.0.0, so the spec needed some changes.

ottomata committed:
Update to service-template-node @ 5cf79c8f1bfab8827b465cd32e6b391505883632

New eslint rules.  The next patch will resolve them.

Conflicts:
	.eslintrc.yml
	.gitignore
	.travis.yml
	app.js
	config.dev.yaml
	config.prod.yaml
	lib/api-util.js
	lib/swagger-ui.js
	lib/util.js
	package.json
	routes/empty.js.template
	routes/ex.js
	routes/v1.js
	spec.template.yaml
	test/features/app/app.js
	test/features/ex/errors.js
	test/features/v1/page.js
	test/features/v1/siteinfo.js
	test/utils/server.js

ottomata committed:
eslint fixes

ottomata commented:
@kostajh

ottomata committed:
More eslint fixes

Pchelolo commented:
Sweet. I even forgot I made some of these changes :) Looks reasonable

ottomata committed:
eslint fix part 3

ottomata committed:
eslint fix part 4

ottomata committed:
eslint fixes part 5

ottomata committed:
eslint fixes part 6

kostajh commented on lib/factories/default-eventgate.js:
```diff
@@ -240,7 +240,7 @@ function makeFileProduce(options, logger, metrics) {
  */
 async function makeKafkaProduce(options, logger, metrics) {
-    // eslint-disable-node/no-missing-require
+    // eslint-disable-node-no-missing-require
```
@ottomata try

```
// eslint-disable-next-line node/no-missing-require
```

that works for me locally.

ottomata committed:
eslint fix part 6

ottomata committed:
Remove node_js 10 from travis build

Just node should suffice, and this
works around an annoying linter / optionalDependency
conflict with node-rdkafka-factory.
